<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DataExplorerDialog</class>
 <widget class="QWidget" name="DataExplorerDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>732</width>
    <height>621</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DataExplorer</string>
  </property>
  <widget class="CodeEditor" name="codeEdit">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>711</width>
     <height>401</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">CodeEditor {
    font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;
    font-size: 8pt;
    color: #000000;
    background-color: #fefff7;
}</string>
   </property>
   <property name="plainText">
    <string>struct Foo {
  u32 x;
  u8 y;
};

struct Cat {
  u16 meow;
};

struct Bar {
  Foo foo;
  u64 z;
};

struct Test {
  u8 a;
  u8 b;
  Cat c;
  Foo foo;
  u8 arr[10];
  Cat c2[2];
};

Test t @ 0x3;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>410</y>
     <width>31</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Log:</string>
   </property>
  </widget>
  <widget class="QTextBrowser" name="logEdit">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>430</y>
     <width>711</width>
     <height>181</height>
    </rect>
   </property>
   <property name="openLinks">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="buttonParse">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>380</y>
     <width>31</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Apple Symbols</family>
     <pointsize>12</pointsize>
     <bold>false</bold>
    </font>
   </property>
   <property name="toolTip">
    <string>Run Pattern</string>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
  color: green;
}</string>
   </property>
   <property name="text">
    <string>▶️</string>
   </property>
   <property name="flat">
    <bool>true</bool>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>CodeEditor.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
